BitList.cs

```diff
@@ -0,0 +1,1553 @@
+// By Lorenzo Lotti
+
+using System;
+using System.Linq;
+using System.IO;
+using System.Collections;
+using System.Collections.Generic;
+
+/// <summary>
+/// Exception generated calling a method when the BitList instance is empty.
+/// </summary>
+public class EmptyInstanceException : Exception { }
+
+/// <summary>
+/// Exception generated when a boolean operation fails.
+/// </summary>
+public class BooleanAlgebraException : Exception { }
+
+/// <summary>
+/// Exception generated when a BitList instance fails to write or read a file.
+/// </summary>
+public class FileException : Exception { }
+
+/// <summary>
+/// Exception generated when a BitList instance fails to read a file.
+/// </summary>
+public class FileReadingException : FileException { }
+
+/// <summary>
+/// Exception generated when a BitList instance fails to write a file.
+/// </summary>
+public class FileWritingException : FileException { }
+
+/// <summary>
+/// Exception generated when a BitList instance fails the conversion to another type.
+/// </summary>
+public class ConversionErrorException : Exception { }
+
+/// <summary>
+/// A list of bits.
+/// </summary>
+/// <exception cref="EmptyInstanceException" />
+/// <exception cref="BooleanAlgebraException" />
+/// <exception cref="FileReadingException" />
+/// <exception cref="FileWritingException" />
+/// <exception cref="ConversionErrorException" />
+public sealed class BitList :
+    IEnumerable<bool>,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['IList<bool>' implements 'ICollection<bool>' so 'ICollection<bool>' can be removed from the inheritance list.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)